Mathlib/Analysis/SpecialFunctions/MulExpNegSq.lean

```diff
@@ -0,0 +1,190 @@
+/-
+Copyright (c) 2024 Jakob Stiefel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jakob Stiefel
+-/
+import Mathlib.Data.Complex.ExponentialBounds
+import Mathlib.Topology.ContinuousMap.Bounded.Basic
+import Mathlib.Topology.MetricSpace.Pseudo.Basic
+
+/-!
+# Definition of `mulExpNegMulSq` and properties
+
+The mapping mulExpNegMulSq transforms a function `g` into another function
+`mulExpNegMulSq g ε : fun x => g x * Real.exp (-ε * g x * g x)`.
+
+## Main Properties
+
+- `mulExpNegMulSq_bounded`: For fixed `ε > 0`, the mapping `mulExpNegMulSq g ε` is bounded by
+  `ε.sqrt⁻¹` uniformly in `g`;
+- `mulExpNegMulSq_tendsto`: For fixed function `g`, the mapping `mulExpNegMulSq g ε`
+  converges pointwise to `g` as `ε → 0`;
+- `continuous_mulExpNegMulSq`: If the function `g` is continuous, the function `mulExpNegMulSq g ε`
+  is continuous;
+- `mulExpNegMulSq_abs_le_norm`: For a fixed bounded continuous function `g`, the mapping
+  `mulExpNegMulSq g ε` is bounded by `norm g`, uniformly in `ε ≥ 0`;
+-/
+
+open NNReal ENNReal BoundedContinuousFunction Filter
+
+section mulExpNegSq
+
+/-! ### Definition and properties of `fun x => x * (Real.exp (- x * x))`-/
+
+/-- Definition of `mulExpNegSq` -/
+noncomputable
+def mulExpNegSq := fun x => x * (Real.exp (- x * x))
```

Comment:
```suggestion
def mulExpNegSq (x : ℝ) := x * exp (- x * x)
```
If you are in the `Real` namespace, no need for `Real.`. Also, some of your parentheses are not necessary and clutter a little bit the expression. Finally, putting the variable left of the colon lets you write its type, making things clearer for the reader.